src/main-process/start.js

```diff
@@ -39,6 +42,12 @@ module.exports = function start (resourcePath, startTime) {
   atomPaths.setUserData(app)
   setupCompileCache()
 
+  const config = getConfig()
```

Comment:
It would be *nice* if we could reuse this `Config` and/or `ConfigFile` instance in `AtomApplication`, in order to avoid reading the config file twice on startup (and also to avoid duplicating the code for locating the config file). That said, I don't think it is necessarily important enough to block merging this PR.